core/src/main/scala/org/broadinstitute/dsde/rawls/entities/compact/CompactEntityProvider.scala

```diff
@@ -240,9 +240,13 @@ class CompactEntityProvider(requestArguments: EntityRequestArguments,
   }
 
   override def listEntities(entityType: String): Source[Entity, NotUsed] =
-    throw new DataEntityException("list all entities not supported for compact data tables.",
-                                  code = StatusCodes.NotImplemented
-    )
+    Source
+      .future(repository.dataSource.inTransaction(ReadOnly) { _ =>
+        repository.queries.listEntities(workspaceId, entityType)
+      })
+      .mapConcat { entityRec =>
+        entityRec.map(_.toEntity).toList
+      }
```

Comment:
This defeats the purpose of streaming. This code says:
* retrieve all entities from the db as a `Seq`, which is fully materialized
* wrap the Seq into a Source

To fully stream, it needs to stay a `Source` all the way from the db to the return value of this function. To implement that, you can't use `dataSource.inTransaction`. See https://github.com/broadinstitute/rawls/blob/d429b0f2727cbf4da1352ccf226f4f10cf5bb8d8/core/src/main/scala/org/broadinstitute/dsde/rawls/entities/compact/entityQuery/EntityQueryStrategy.scala#L33 for an example of generating a `Source` directly from a database query.